polls/views.py

```diff
@@ -436,3 +437,32 @@ def post(self, request):
 
 class TemporaryLoginTemplate(TemplateView):
     template_name = "polls/temp_login_template.html"
+
+
+class CreateExperimentTypeTaskPresentationView(
+    LoginRequiredMixin, UserPassesTestMixin, View
+):
+    def get(self, request):
+        experiment_types = ExperimentType.objects.all()
+        context = {"experiment_types": experiment_types}
+        return render(request, "polls/experiment_type_task_presentation.html", context)
+
+    def post(self, request, *args, **kwargs):
+        exp_type = request.POST.get("experiment_type")
+        task_presentation = request.POST.get("task_presentaion")
+        experiment_type = ExperimentType.objects.get(pk=exp_type)
+
```

Comment:
This would fail if an id that doesn't exist is supplied, django would auto return 500 and the rest of the code wouldn't be executed

Reply:
But as we are using a dropdown, which will only show the type that exist in db, I do not think that such a case will be there. Because it is not an API it is a template view I guess we can rely on dropdown. but we can change it if necessary.

Reply:
But I have added try catch for error handling. Please have a look

Reply:
True, we're using the dropdown, but we still have to prepare cause we should expect all types of users, an expert can easily boycott our dropdown and send whatever he likes to the db, hence we should prepare for such people as well

Reply:
Got it. I have used try catch for this